src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,156 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { getPeople } from '../api';
+
+function getPreperedPeople(
+  filterField: string,
+  query: string,
+  centuries: string[],
+  sort: string,
+  sortOrder: string,
+  people: Person[],
+) {
+  let visiblePeople = [...people];
+
+  const preperedQuery = query.trim().toLowerCase();
+
+  visiblePeople = visiblePeople.filter(
+    person =>
+      person.name.toLowerCase().includes(preperedQuery) ||
+      person.motherName?.toLowerCase().includes(preperedQuery) ||
+      person.fatherName?.toLowerCase().includes(preperedQuery),
+  );
+
+  if (centuries.length > 0) {
+    visiblePeople = visiblePeople.filter(person =>
+      centuries.includes(Math.ceil(person.born / 100).toString()),
+    );
+  }
+
+  if (sort) {
+    visiblePeople.sort((a: Person, b: Person) => {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          if (sortOrder === 'asc') {
+            return a[sort].localeCompare(b[sort]);
+          } else if (sortOrder === 'desc') {
+            return b[sort].localeCompare(a[sort]);
+          } else {
+            return 0;
+          }
+
+        case 'born':
+        case 'died':
+          if (sortOrder === 'asc') {
+            return a[sort] - b[sort];
+          } else if (sortOrder === 'desc') {
+            return b[sort] - a[sort];
+          } else {
+            return 0;
+          }
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  switch (filterField) {
+    case 'all':
+      return visiblePeople;
+    case 'm':
+      return visiblePeople.filter(person => person.sex === 'm');
+    case 'f':
+      return visiblePeople.filter(person => person.sex === 'f');
+    default:
+      return visiblePeople;
+  }
```

Comment:
```suggestion

  if (filterField) {
     visiblePeople = visiblePeople.filter(person => person.sex === filterField);
  }
  
  return visiblePeople;
```
we can simplify that